build.psm1

```diff
@@ -3043,6 +3043,10 @@ function New-TestPackage
     # Build test tools so they are placed in appropriate folders under 'test' then copy to package root.
     $null = Publish-PSTestTools
     $powerShellTestRoot =  Join-Path $PSScriptRoot 'test'
+
+    # Install SelfSignedCertificate module from gallery
+    Save-Module -Name SelfSignedCertificate -Path "$powerShellTestRoot/tools/Modules" -Repository "PSGallery" -MinimumVersion '0.0.2' -Force -Confirm:$false
```

Comment:
Might be worth `RequiredVersion` rather than `MinimumVersion`. I don't want to cause a build breakage if I push a new version that doesn't work.

Reply:
Actually using a MinimumVersion would mean at least 0.0.2. When I actually run it, it gets me 0.0.3.

Reply:
If we do pin it, we should do something like we do for the modules we use for PSModuleRestore, use a csproj, and dependabot to submit PRs to update the csproj

Reply:
I think we should not pin it to a specific version and make sure that updates to the module are backwards compatible.

Reply:
> Actually using a MinimumVersion would mean at least 0.0.2. When I actually run it, it gets me 0.0.3.

Right. My point is that the module isn't particularly well maintained or tested, and PowerShell won't want or depend on any new features from it. I only see cons in not pinning it. Changing the version used should be a conscious decision.

Reply:
Especially since it's v0.x (i.e. no guarantee of stability under semver)

Reply:
I do not want to miss on the bug fixes / features future versions on this module might have. Also, in all likelihood if there are any changes, the WebListener tests will be updated too. If I pin to a specific version then the tests might not work.

Reply:
Talked to @adityapatwardhan offline. Some considerations are:
- I own the module (not Microsoft)
- It's simple script that wraps .NET crypto API calls in as minimal a fashion as possible
- A bugfix as a "pull" from PowerShell Core rather than as a "push" from the module is the most likely scenario

Given these points, plus the caveats of unpinned dependencies, we agreed to pin the version and use the csproj strategy used for the src build [here](https://github.com/PowerShell/PowerShell/blob/master/src/Modules/PSGalleryModules.csproj).